logstash-core/src/main/java/org/logstash/secret/store/SecretStoreFactory.java

```diff
@@ -27,29 +25,73 @@ private SecretStoreFactory() {
 
     private static final Logger LOGGER = LogManager.getLogger(SecretStoreFactory.class);
 
+    private enum MODE {LOAD, CREATE, EXISTS, DELETE}
+
+    /**
+     * Determine if this secret store currently exists
+     */
+    public static boolean exists(SecureConfig secureConfig) {
+        return doIt(MODE.EXISTS, secureConfig).exists(secureConfig);
+    }
+
+    /**
+     * Creates a new {@link SecretStore} based on the provided configuration
+     *
+     * @param secureConfig The configuration to pass to the implementation
+     * @return the newly created SecretStore
+     * @throws {@link SecretStoreException} if errors occur while loading, or if store already exists
+     */
+    static public SecretStore create(SecureConfig secureConfig) {
+        return doIt(MODE.CREATE, secureConfig);
+    }
+
     /**
-     * Creates a {@link SecretStore} based on the provided configuration
+     * Deletes a {@link SecretStore} based on the provided configuration
      *
      * @param secureConfig The configuration to pass to the implementation
-     * @return
+     * @throws {@link SecretStoreException} if errors occur
      */
+    static public void delete(SecureConfig secureConfig) {
+        doIt(MODE.DELETE, secureConfig);
+    }
+
+    /**
+     * Loads an existing {@link SecretStore} based on the provided configuration
+     *
+     * @param secureConfig The configuration to pass to the implementation
+     * @return the loaded SecretStore
+     * @throws {@link SecretStoreException} if errors occur while loading, or if store does not exist
+     */
+    static public SecretStore load(SecureConfig secureConfig) {
+        return doIt(MODE.LOAD, secureConfig);
+    }
+
     @SuppressWarnings({"unchecked", "JavaReflectionMemberAccess"})
-    static public SecretStore loadSecretStore(SecureConfig secureConfig) {
-        //cheap SPI, if we ever support more then one implementation we should expose it as a setting and push the class name here via the secureConfig
-        String className = System.getProperty("org.logstash.secret.store.SecretStore", "org.logstash.secret.store.backend.JavaKeyStore");
```

Comment:
The system property was removed via [the prior review comment](https://github.com/elastic/logstash/pull/8659#discussion_r152057155)